### PR TITLE
Group views.activity into classes

### PR DESCRIPTION
**Pull request by seanh**
Group the view functions in h.views.activity into classes.

This doesn't make any behavioural changes to the code, just:

* Groups the views into classes, reducing some duplication and
  boilerplate, especially in view configuration.

* The check for whether the feature flag is enabled now only needs to be
  done in one place.

* Lots of references to request become self.request

* Some functions renamed, e.g. group_search() -> GroupSearchController.search()

* Group and user search use super() to call the core search view

* The tests are arranged so that it's now on test class per view
  controller class. In some cases this means multiple methods are tested
  in one class, previously it was one test class per function to test

### CODE REVIEW

**Comment by seanh** [comment]
It would be good if this could be reviewed quickly, since anything that changes `h/views/activity.py` is going to clash with this. The diff is huge because it indents and moves around a lot of code - it doesn't actually make any behavioral changes, just organizing into classes.

I need to make some changes to `h.views.activity`, especially to the routing configuration, to do <https://github.com/hypothesis/h/issues/4004>. The way `h.views.activity` is on master makes this a pain to do and will mean that I'll only be making the mess worse. So this pull request makes the necessary changes easier to make, by reducing boilerplate and duplication, especially in the view configurations.

**Comment by robertknight** [comment]
I'm still looking through this at the moment but an initial concern I have is that it binds the functionality of the user and group sidebars ever tighter into the `/groups/<pubid>/` and `/users/<username>` routes, which I don't think was the intent of the design.

I wonder if it would make more sense to pull the user and group sidebars into their own panels which are just rendered from activity pages via `{{ panel(...) }}` calls, similar to what we do with the navbar.

File: tests/h/views/activity_test.py [hunk 1]
```diff
@@ -344,182 +519,211 @@ def test_it_redirects_to_user_search(self, pyramid_request):
...
 
+    @pytest.mark.usefixtures('delete_lozenge_request')
+    def test_delete_lozenge_returns_a_redirect(self, controller):
+        result = controller.delete_lozenge()
 
+        assert isinstance(result, httpexceptions.HTTPSeeOther)
 
-@pytest.mark.usefixtures('groups_service', 'routes')
-class TestGroupLeave(object):
+        # This tests that the location redirect to is correct and also that
```
Comment (robertknight) [inline tests/h/views/activity_test.py]: Typo "redirect to is correct"

**Comment by robertknight** [comment]
Given the size of the diff I haven't examined every line closely and I still think it might be worth looking into extracting the sidebars out into their panels in future. LGTM.

**Pull request merged**